extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/CdiAwareJob.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.quartz.runtime;
+
+import jakarta.enterprise.context.Dependent;
+import jakarta.enterprise.inject.Instance;
+
+import org.quartz.Job;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.Scheduler;
+import org.quartz.spi.TriggerFiredBundle;
+
+/**
+ * An abstraction allowing proper destruction of Job instances in case they are dependent beans.
+ * According to {@link org.quartz.spi.JobFactory#newJob(TriggerFiredBundle, Scheduler)}, a new job instance is created for every
+ * trigger.
+ * We will therefore create a new dependent bean for every trigger and destroy it afterwards.
+ */
+public class CdiAwareJob implements Job {
+
+    final private Instance.Handle<? extends Job> jobInstanceHandle;
```

Comment:
I believe that the convention is to use `private final` but that's just nitpicking ;-)

Reply:
Done